Makefile

```diff
@@ -226,18 +232,38 @@ release-image-tag: ## Prints the image tag of the release image
 ## End-to-end (E2E) Testing ##
 ##############################
 
-e2e-setup: ## Setup end-to-end tests
-	@cd e2e && npm install
-	@cd e2e && npx playwright install --with-deps
+e2e-build: ## Build the e2e Docker image, if not already built, using ./e2e/Dockerfile
+	@if [ -z "$$(docker images -q playwright-e2e)" ]; then \
+	  echo "Building Docker image..."; \
+	  docker build -t playwright-e2e -f ./e2e/Dockerfile .; \
+	else \
+	  echo "Docker image already exists, skipping build."; \
+	fi
+
+e2e-copy-report: ## Copy the Playwright report from the container to local
+	docker cp playwright-e2e-container:/e2e/playwright-report ./e2e/playwright-report
+
+e2e-clean-report: ## Remove the local ./e2e/playwright-report folder and its contents
+	rm -rf ./e2e/playwright-report
+
+e2e-delete-image: ## Delete the Docker image for e2e tests
+	@docker rmi -f playwright-e2e 2>/dev/null || echo "Docker image playwright-e2e does not exist, skipping."
 
-e2e-setup-ci: ## Install system dependencies, Node dependencies, and Playwright browsers
-	sudo apt-get update
-	sudo apt-get install -y libwoff1 libopus0 libvpx7 libevent-2.1-7 libopus0 libgstreamer1.0-0 \
-	libgstreamer-plugins-base1.0-0 libgstreamer-plugins-good1.0-0 libharfbuzz-icu0 libhyphen0 \
-	libenchant-2-2 libflite1 libgles2 libx264-dev
-	cd e2e && npm ci
-	cd e2e && npx playwright install --with-deps
+e2e-run: ## Run the Playwright tests in a Docker container and copy the report locally
+e2e-run: e2e-build
+	@:$(call check_defined, APP_NAME, You must pass in a specific APP_NAME)
+	@:$(call check_defined, BASE_URL, You must pass in a BASE_URL)
+	docker rm -f playwright-e2e-container || true
+	docker run --name playwright-e2e-container -e APP_NAME=$(APP_NAME) -e BASE_URL=$(BASE_URL) playwright-e2e
+	$(MAKE) e2e-clean-report
+	$(MAKE) e2e-copy-report
```

Comment:
Rather than separately copying the report from the container file system, could consider mounting a location from the host system in the `docker run` call for playwright to write its report to then can update to use `docker run --rm` to keep the containers cleaned up from each run (rather than having to separately call `docker rm` up above as well).